### DISCUSSION TITLE
Search only returns the first 5000 results

### DISCUSSION THREAD

**Discussion by chengguangnan** [immich-app/immich#9976] [Q&A] [answered]
Hi, I want to search/filter and move the results into an album, but the result page seems only return the first 5000 items. 

I could repeat this search again with option "Not in any albumn" to get the next 5000 items. However I want to know if this experience  could be improved.

**Comment by mertalev** [comment, marked as answer]
The 5000 limit is a web limitation to avoid browser jank. If you have programming knowledge, you can use the API directly instead. See [searchMetadata](https://immich.app/docs/api/search-metadata) and [searchSmart](https://immich.app/docs/api/search-smart).
